### DISCUSSION TITLE
Checks from another PR are applied to existing PR

### DISCUSSION THREAD

**Discussion by jdnvn** [community/community#57632] [Actions]
### Select Topic Area

Bug

### Body

I have an open PR from my `release` branch targeting my `master` branch, which should have its own CI checks. However, when I create a PR from the `release` branch targeting my `develop` branch, the actions kick off and are applied to the `release->master` PR checks. This is problematic because the `develop` branch has upstream changes that master doesn't have, so the tests may fail in the `release->develop ` merge branch but not the `release->master` merge branch and it will still look like `release->master` failed. We also have these actions as required checks for master, so this could prevent or allow merging when CI is not accurate.

TLDR:

Open PR from `release` to `master` with passing CI

PR opened from `release` to `develop` fails CI

Original `release->master` PR can no longer be merged because it *thinks* PR is failing from `release->develop` PR.
